controller/linodemachine_controller_helpers.go

```diff
@@ -74,8 +80,24 @@ func (r *LinodeMachineReconciler) newCreateConfig(ctx context.Context, machineSc
 
 		return nil, err
 	}
-	createConfig.Metadata = &linodego.InstanceMetadataOptions{
-		UserData: b64.StdEncoding.EncodeToString(bootstrapData),
+
+	region, err := machineScope.LinodeClient.GetRegion(ctx, machineScope.LinodeMachine.Spec.Region)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
nit: Why is an error here allowed to stop the controller from reconciling the creation of a LinodeMachine? If this fails, we can gracefully handle this by just trying the Metadata service anyways and falling back on the StackScripts if it fails.

Reply:
we need to know before linode creation if we are using metadata or stackscripts so we can't fail back to anything once the linode is created

Reply:
nit: Now that we have removed the `LinodeMachine.spec.UseStackScriptBootstrap` setting, can we gracefully handle these two API call errors here instead of just returning them? 

We can default to setting the `createConfig.StackScript*` fields.